feat: attach __webpack_nonce__ to inline <style> if set

dominics opened the pull request:
This allows Content Security Policies to use a nonce, in combination with disallowing "unsafe-inline" changes to the page, giving better protection against injection.

Tests incoming

mathio reviewed (commented):
Nice 👍 We should also [update embed-react README file](https://github.com/Typeform/embed/tree/main/packages/embed-react). I can do this on Monday for you and then we can merge the PR.

mathio commented:
There seems to be an issue with the Github Action, I am looking into it. 

@dominics I just realized since this is a PR from your fork, I can not push any changes to it. Can you please add short section about this into react SDK README file? [Into "Configuration" section, above the "Examples" section](https://github.com/Typeform/embed/tree/main/packages/embed-react#configuration). You can just mention the feature and link to #458 for more details.

mathio commented:
@dominics the issue with Github Actions seems to be resolved now, I tested with a dummy PR from a fork via non-Typeform Github account. Please rebase on latest `main` branch.

dominics commented:
Rebased on main, and added [a short section to the README](https://github.com/Typeform/embed/pull/457/commits/e3d16c408b0f594d4c08298b8fc46d4a25a35fee#diff-ea8942465faabd2ed7f995ccd7ae9211b04dfb67012ecaf25ae15baae9758184R90) 👍

typeform-ops-gha commented:
:tada: This PR is included in version @typeform/embed-react-v1.3.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@typeform/embed-react/v/1.3.0)
- [GitHub release](https://github.com/Typeform/embed/releases/tag/%40typeform/embed-react-v1.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
